pkg/apis/cluster/v1alpha1/cluster_types_test.go

```diff
@@ -75,6 +75,10 @@ func crudAccessToClusterClient(t *testing.T, cs *clientset.Clientset) {
 	}
 
 	// Test deleting the item for delete requests.
+	actual.Finalizers = nil
+	if _, updateErr := client.Update(actual); updateErr != nil {
```

Comment:
Can you explain why this was added to the test and what it is testing? I don't see a new Update method, I don't see any verification that the update did what it was supposed to do other than not return an error.

Reply:
I had to delete the finalizer on the cluster in order for it to actually delete the Cluster object itself. This test is making sure that we can delete the Cluster object, but that is prevented by the existence of the finalizer. This is the same pattern followed by [machine_types_test](https://github.com/mkjelland/cluster-api/blob/25e80c650d0fb0c187d5eb50cb13499f2c302dea/pkg/apis/cluster/v1alpha1/machine_types_test.go)

Reply:
Ok - I don't quite understand why a client would need to know about setting the Finalizers (it seems like a detail that should be hidden), but I'll trust this is the right thing to do or that it is following a previous pattern we don't want to change right now.

Reply:
This test is making sure that the object can be deleted but ignoring the cluster controller's functionality. The test broke when I made these changes because the finalizer prevents the object from being deleted. 

The point of the finalizer is to allow a controller to take some action before the Kubernetes object disappears. In this case, the cluster controller sees the delete action, then it calls the actuator's delete function, if the delete action succeeds, the controller itself deletes the finalizer and the object will be deleted. The reason for this is that the actuator needs to access the object in order to know how to delete it. Outside of the controller, no one should need to delete the finalizer from the cluster.